saleor/app/manifest_validations.py

```diff
@@ -188,47 +186,21 @@ def clean_extensions(manifest_data, app_permissions, errors):
 def clean_webhooks(manifest_data, errors):
     webhooks = manifest_data.get("webhooks", [])
 
-    async_types = {
-        str_to_enum(e_type[0]): e_type[0] for e_type in WebhookEventAsyncType.CHOICES
-    }
-    sync_types = {
-        str_to_enum(e_type[0]): e_type[0] for e_type in WebhookEventSyncType.CHOICES
-    }
-
     target_url_validator = AppURLValidator(
         schemes=["http", "https", "awssqs", "gcpubsub"]
     )
 
     for webhook in webhooks:
-        if not validate_subscription_query(webhook["query"]):
+        subscription_query = SubscriptionQuery(webhook["query"])
+        if not subscription_query.is_valid:
             errors["webhooks"].append(
                 ValidationError(
-                    "Subscription query is not valid.",
+                    "Subscription query is not valid: " + subscription_query.error_msg,
                     code=AppErrorCode.INVALID.value,
                 )
             )
 
-        webhook["events"] = []
-        for e_type in webhook.get("asyncEvents", []):
-            try:
-                webhook["events"].append(async_types[e_type])
-            except KeyError:
-                errors["webhooks"].append(
-                    ValidationError(
-                        "Invalid asynchronous event.",
-                        code=AppErrorCode.INVALID.value,
-                    )
-                )
-        for e_type in webhook.get("syncEvents", []):
-            try:
-                webhook["events"].append(sync_types[e_type])
-            except KeyError:
-                errors["webhooks"].append(
-                    ValidationError(
-                        "Invalid synchronous event.",
-                        code=AppErrorCode.INVALID.value,
-                    )
-                )
+        webhook["events"] = subscription_query.events
```

Comment:
With this change, we don't need `asyncEvents` and `syncEvents` fields at all. If it is ok, where can I find some description to mention that?

Reply:
I think we can deprecate these fields. In the description we could say that these fields can be skipped whenever the `query` param is passed. But that would also mean that we would deprecate the old static payloads in favor of subscription ones.

Reply:
Hmm, but do we support subscriptions for EACH webhook there is in Saleor? Only then we could deprecate these input fields.